tests/test_dependencies.py

```diff
@@ -471,37 +478,59 @@ def test_middleware_response_without_content_type(self):
         request = Mock()
         self.assertEqual(response, middleware(request=request))
 
-    def test_middleware_response_with_components_with_slash_dash_and_underscore(
-        self,
-    ):
+    def test_middleware_response_with_components_with_slash_dash_and_underscore(self):
         registry.register("dynamic", DynamicComponent)
+        registry.register("test-component", component=SimpleComponent)
+        registry.register("test/component", component=SimpleComponent)
+        registry.register("test_component", component=SimpleComponent)
 
-        component_names = [
-            "test-component",
-            "test/component",
-            "test_component",
-        ]
-        for component_name in component_names:
```

Comment:
I refactored this test to remove the for loop. The reason being that in each loop iteration, the resulting HTML would have different component IDs, so it was hard to make sense of which IDs were expected in which iteration.